pkg/atomfs/molecule.go

```diff
@@ -24,6 +24,19 @@ type Molecule struct {
 // mountUnderlyingAtoms mounts all the underlying atoms at
 // config.MountedAtomsPath().
 func (m Molecule) mountUnderlyingAtoms() error {
+	// in the case that we have a verity or other mount error we need to
+	// tear down the other underlying atoms so we don't leave verity and loop
+	// devices around unused.
+	atomsMounted := []string{}
+	cleanupAtoms := func(err error) error {
+		for _, target := range atomsMounted {
+			if err := squashfs.Umount(target); err != nil {
+				return errors.Wrapf(err, "failed to unmount atom @ target %q while handling error: %s", target, err)
```

Comment:
i think you've stomped on 'err' passed in here. so your wrapped 'err' is the *new* err, not the one passed in.

You should do something like:

```go
   if umountErr := squashfs.Umount(target); umountErr != nil {
    return errors.Wrapf(err, "failed to unmount atom @ target %q while handling error: %s", target, umountErr)
   }
```

I think that is better, but then do you want to return on the error or just collect it and continue trying to unmount the other targets ?

Reply:
Note, in the case of an unmount error, I'll wrap the umountErr, but include out the contents of the passed in err in the wrapped message.

```golang
if umountErr := squashfs.Umount(target); umountErr != nil {
    return errors.Wrapf(umountErr, "failed to unmount atom @ target %q while handling error: %s", target, err)
}
```